pkg/pubsub/dapr/dapr.go

```diff
@@ -0,0 +1,83 @@
+package dapr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dapr/go-sdk/client"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+)
+
+type ClientConfig struct {
+	// Name of the component to be used for pub sub messaging
+	Component string `json:"component"`
+}
+
+// Dapr represents driver for interacting with pub sub using dapr.
+type Dapr struct {
+	// Array of clients to talk to different endpoints
+	client client.Client
+
+	// Name of the pubsub component
+	pubSubComponent string
+}
+
+const (
+	Name = "dapr"
+)
+
+func (r *Dapr) Publish(ctx context.Context, data interface{}, topic string) error {
```

Comment:
can we add unit tests for this? can be a follow-up

Reply:
We must incorporate a mock dapr client/server for unit testing of these functions. Follow-up will be better to keep the changes small.